fix(deploy): allow the container to raise in MIGs

gustavovalverde opened the pull request:
## Motivation

The Managed Instance Groups in GCP were not raising the container as those were expecting the device name and the correct RW/RO mode to match between the created disk and the disk we're mounting in the container.

This also fixes monitoring, and uses the same caching directory as in CI, as the Container Optimized Image does not allow mounting in non-standard linux paths for security reasons.

Fixes: #5644
Fixes: #6861  

### Specifications

See: https://cloud.google.com/sdk/gcloud/reference/compute/instance-templates/create-with-container#--container-mount-disk

## Solution

- Add the missing `--boot-disk-size 300GB` to the MIGs template, just as we have on CI and Single instance deploy
- Allow user output and container `stdin` and `tty` to standardize with other approaches
- Enable Google Monitoring for single and long-live instances (MIGs)
- Use `ZEBRA_CACHED_STATE_DIR` with `/var/cache/zebrad-cache` as we do on CI, as the `cos-cloud` doesn't behave correctly with non-standard paths.

## Review

Anyone from DevOps

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

## Follow Up Work

We should make further changes in a following PR:

- For releases, it should sync from scratch (1). Currently it's 2-4 weeks between releases, so a 3 day mainnet sync is ok.

- For `main` branch pushes, it should have a cached state (2 or 3, whatever is easier or more reliable). We merge new PRs multiple times a day, so a cached state is needed.

gustavovalverde committed:
fix(deploy): allow the container to raise in MIGs

teor2345 commented on .github/workflows/continous-delivery.yml:
```diff
@@ -211,15 +211,18 @@ jobs:
-          --container-env "NETWORK=${{ matrix.network }},LOG_FILE=${{ vars.CD_LOG_FILE }},SENTRY_DSN=${{ vars.SENTRY_DSN }},SHORT_SHA=${{ env.GITHUB_SHA_SHORT }}" \
-          --create-disk=name=zebrad-cache-${{ env.GITHUB_SHA_SHORT }}-${NETWORK},auto-delete=yes,size=300GB,type=pd-ssd \
-          --container-mount-disk=mount-path="/zebrad-cache",name=zebrad-cache-${{ env.GITHUB_SHA_SHORT }}-${NETWORK} \
+          --container-env "NETWORK=${{ matrix.network }},ZEBRA_CACHED_STATE_DIR=/var/cache/zebrad-cache,LOG_FILE=${{ vars.CD_LOG_FILE }},SENTRY_DSN=${{ vars.SENTRY_DSN }},SHORT_SHA=${{ env.GITHUB_SHA_SHORT }}" \
```
Using a different state path will need changes in `runtime-entrypoint.sh`. You should just be able to substitute the variable in directly, like this:
https://github.com/ZcashFoundation/zebra/blob/286ebdbf8c9e40891164821852567bc86de30012/docker/runtime-entrypoint.sh#L39

It would also be nice to set this path as the default in the `Dockerfile`, to avoid confusion.

If you're looking for an example of how runtime config creation works, see PR #6890. (But terminal colour is a bit more complicated because the variable doesn't match the config setting.)

gustavovalverde replied:
Solved with: [fix(docker): add the ZEBRA_CACHED_STATE_DIR as a default ENV](https://github.com/ZcashFoundation/zebra/pull/6893/commits/10c3fc9cc6e6a813f20a8f9074b108b53b9d5936)

teor2345 commented:
> * For releases, it should sync from scratch (1). Currently it's 2-4 weeks between releases, so a 3 day mainnet sync is ok.

These further changes could be optional. For testing, it's enough to just run a scheduled full sync in our test environment. To support the network, using a cached state will be better because the instances are live faster.

gustavovalverde committed:
fix(docker): add the `ZEBRA_CACHED_STATE_DIR` as a default `ENV`

This no longer requires the env variable to be defined in other places, unless we're changing the default configuration

gustavovalverde committed:
Merge branch 'main' into fix-migs-logging

dconnolly reviewed (approved):
Approving; interesting if mode=rw matters, docs say it is the default when unspecified for both `--create-disk` and `--container-mount-disk`
